File: dojo_9/java/src/main/java/com/mercadolibre/dojos/Tramo.java
package com.mercadolibre.dojos;

public class Tramo {
    private final Transporte transporte;
    private final Lugar destino;
    private final Lugar origen;

    public Tramo(Lugar origen, Lugar destino, Transporte transporte) {
        this.transporte = transporte;
        this.origen = origen;
        this.destino = destino;
    }

    public Precio precio() {
        return transporte.precio(origen, destino);
    }
}

File: dojo_kotlin/app/src/main/java/com/nelsito/kotlindojo9/Tramo.kt
package com.nelsito.kotlindojo9

/**
 * Created by npuglisi on 12/12/17.
 */
class Tramo(val origen: Lugar, val destino: Lugar, val transporte: Transporte) {
    fun precio(): Precio {
        return transporte.precio(origen, destino)
    }
}